lua/lualine/utils/utils.lua

```diff
@@ -7,17 +7,35 @@ local M = {}
 -- functionality at runtime .
 function M.expand_set_theme(func)
   -- execute a local version of global function to not get in a inf recurtion
-  local set_theme = _G.set_lualine_theme
-  _G.set_lualine_theme = function()
+  local set_theme = _G.lualine_set_theme
+  _G.lualine_set_theme = function()
     set_theme()
     func()
   end
 end
 
 -- Note for now only works for termguicolors scope can be background or foreground
 function M.extract_highlight_colors(color_group, scope)
-  local color = string.format('#%06x', vim.api.nvim_get_hl_by_name(color_group, true)[scope])
-  return color or nil
+  if vim.fn.hlexists(color_group) == 0 then return nil end
+  local gui_colors = vim.api.nvim_get_hl_by_name(color_group, true)
+  local cterm_colors = vim.api.nvim_get_hl_by_name(color_group, false)
+  local color = {
+    ctermfg = cterm_colors.foreground,
+    ctermbg = cterm_colors.background,
+  }
+  if gui_colors.background then
+    color.guibg = string.format('#%06x', gui_colors.background)
+    gui_colors.background = nil
+  end
+  if gui_colors.foreground then
+    color.guifg = string.format('#%06x', gui_colors.foreground)
+    gui_colors.foreground = nil
+  end
+  cterm_colors.background = nil
+  cterm_colors.foreground = nil
```

Comment:
why is `cterm_colors` reset to nil here?

Reply:
I'm freeing unessery values so vim.tbl_extend doesn't put them in . They have already been loaded in the list  with ctermfg and ctermbg name

Reply:
I don't get it. What's the point of cterm_colors then? Seems to me like you get it just to clear it...

Reply:
I'm returning a table with
{ctermfg, ctermbg, guifg, guibg} + what ever parameters are set by gui and cterm sections (if available).

Reply:
in line 23-25 I've alrwady loaded those values to The tavle I'll return(color)

Reply:
then why do we have to extend `cterm_colors` if it contains nothing? Won't this yield the same results?
```lua
  color = vim.tbl_extend('keep', color, gui_colors)
```

Reply:
>then why do we have to extend cterm_colors if it contains nothing? Won't this yield the same results?

cterm_colors table can cointain other flags like bold=true, reverse=true . That's why I'm useing tbl_extend. And removing the foreground and background entry as they have already been included in different name.